include/swift/Sema/ConstraintSystem.h

```diff
@@ -1386,6 +1386,13 @@ class Solution {
   /// "resolved" concrete type.
   Type getResolvedType(ASTNode node) const;
 
+  /// Retrieve the type of the given node as recorded in this solution
+  /// and resolve all of the type variables in contains to form a fully
+  /// "resolved" concrete type.
+  /// If no type has been computed for this node or if \c node is null, returns
+  /// the null type.
+  Type getResolvedTypeIfAvailable(ASTNode node) const;
```

Comment:
I don't see any uses of this but I'd say that if ASTNode doesn't have a type it's either not associated with the solution which is an issue on the caller side, or it didn't get recorded during constraint generation, which makes it a bug in the solver.

Reply:
Oh, looks like the need for that method has gone away at some point during my development. I just removed it.